Proton/Sources/Core/RichTextView.swift

```diff
@@ -372,17 +372,13 @@ class RichTextView: AutogrowingTextView {
     }
 
     func insertAttachment(in range: NSRange, attachment: Attachment) {
+        richTextStorage.beginEditing()
         richTextStorage.insertAttachment(in: range, attachment: attachment)
         if let rangeInContainer = attachment.rangeInContainer() {
-            edited(range: rangeInContainer)
+            richTextStorage.edited([.editedCharacters, .editedAttributes], range: rangeInContainer, changeInLength: 0)
         }
-        scrollRangeToVisible(NSRange(location: range.location, length: 1))
-    }
-
-    func edited(range: NSRange) {
-        richTextStorage.beginEditing()
-        richTextStorage.edited([.editedCharacters, .editedAttributes], range: range, changeInLength: 0)
         richTextStorage.endEditing()
```

Comment:
Calling `richTextStorage.insertAttachment(...)` will trigger a relayout. 
Calling `richTextStorage.endEditing()` will trigger a relayout.
I've merged these two methods and wrapped the whole thing in a single `beginEditing()` and `endEditing()` transaction.